code_1.rb

```diff
@@ -0,0 +1,104 @@
+class Station # класс станция
+  attr_reader :trains, :name_station # возвращает список поездов, находящися на станции
+  def initialize(name_station) # имя станции указывается при создании
+    @name_station = name_station
+    @trains = [] # массив поездов
+  end
+  def add_trains (train) # принимает поезда по одному за раз
+    self.trains << train
+  end
+  def delete_trains (train) # отправляет поезда по одному за раз
+    self.trains.delete(train)
+  end
+  def type_trains # возвращает список поездов на станции по типу: кол-во грузовых, пассажрских
+    kol_gr = 0
+    kol_ps = 0
+    puts "Грузовые: "
+    for tr in self.trains
+      if tr.type == 'gr'
+        kol_gr += 1
+        puts tr
+      end
+    end
+    puts "Пассажирские: "
+    for tr in self.trains
+      if tr.type == 'ps'
+        kol_ps += 1
+        puts tr
+      end
+    end
+    puts "Количество пассажирских #{kol_ps}"
+    puts "Количество грузовых #{kol_gr}"
+  end
+end
+
+class Route # класс маршрут
+  attr_reader :stations, :start, :finish # возвращает список всех станций маршрута
+  def initialize (start, finish) # начальная и конечная станция указываются при создании
+    @start = start
+    @finish = finish
+    @stations = [start, finish] #массив станций
+  end
+  def print # выводит список всех станций по-порядку от начальной до конечной
+    puts self.stations
```

Comment:
при обращении к геттеру `self` не нужен
```suggestion
    puts stations
```

Reply:
но вообще при выводе информации более верно выводить не объекты целиком, а какой-то (требуемый) аттрибут
например имя каждой из станций